frontend/packages/ceph-storage-plugin/src/components/kms-config/kms-config.tsx

```diff
@@ -42,7 +42,7 @@ export const KMSConfigure: React.FC<KMSConfigureProps> = ({
 
   const isHpcsKmsSupported = useFlag(GUARDED_FEATURES.ODF_HPCS_KMS);
   // vault as default KMS
-  const kmsProvider: ProviderNames = state.kms?.['kmsProvider'] || ProviderNames.VAULT;
+  const kmsProvider: ProviderNames = state.kms?.['provider'] || ProviderNames.VAULT;
```

Comment:
```suggestion
  const kmsProvider: ProviderNames = state.kms?.provider || ProviderNames.VAULT;
```
Let's use array indexing if the key is dynamic.

Reply:
here ```kms``` type is ```VaultConfig | KMSConfig```.   only  KMSConfig has a "provider" key. 
```state.kms?.provider ``` will give error
VaultConfig is an older type, we are keeping it for backward compatibility (it will be removed once we remove the older storage cluster creation wizard flow).